lib/src/main/kotlin/com/swmansion/starknet/data/serializers/BlockIdSerializer.kt

```diff
@@ -19,10 +19,14 @@ internal object BlockIdSerializer : KSerializer<BlockId> {
         val value = decoder.decodeString()
 
         if (BlockTag.entries.map { it.tag }.contains(value)) {
-            val tag = BlockTag.valueOf(value)
+            val tag = BlockTag.fromValue(value)
```

Comment:
Standard `valueOf()` takes enum's constant, while we want to retrieve enum's constant based on the value.